Added IRC support, since it didn't even show up with this mod and a rank

Miniontoby opened the pull request:
I have added IRC support, since it didn't even show up with this mod and a rank

and I also updated the mod.conf and remove the old deprecated files

Miniontoby committed:
Added IRC support, since it didn't even show up with this mod and a rank

octacian commented:
Thanks so much for the contribution! One question: is `[off]` currently a standard in the Minetest community to disable forwarding to IRC? If it's not used elsewhere already, could we maybe come up with something less verbose?

Miniontoby commented:
What do you mean with "is [off] currently a standart in the Minetest community to disable forwarding to IRC"?

Btw I think I will have to fix the code a bit, since if you do not have a rank it will still show the block quotes. And yeah it might be better to not show the rank at IRC

EDIT: You mean `or message:sub(1, 5) == "[off]"`. Uhmm this is just copied from the IRC mod

octacian commented:
Oh okay so in that case it sounds like it is a standard then. It looks like it disables forwarding any messages to IRC that start with `[off]`. A bit annoyingly verbose, but if it's a standard, I'll take it :smile:

Let me know when you've fixed & tested things :)

Miniontoby committed:
Removed prefix from irc

Because there is no good way to add a prefix before the name without changing the irc.playerMessage function.

Miniontoby commented:
I have remove the prefix for IRC for now. 

It could be done, but it would look like `<[Admin] Miniontoby> Hello` or `<Miniontoby> [Admin] Hello`, which will breaks bots.

So it is up to you if you want to do that, but for now I would say its fine, because if the IRC mod would have just worked, then that also wouldn't have showed up.

octacian commented:
I mean if that's the only way to make it work, I'm happy to concede. Although, do you mind explaining how it breaks bots? Especially how `<Miniontoby> [Admin] Hello` or `<Admin:MinionToby> Hello`  or something else similar would break a bot.

I'm not super familiar with IRC anymore, it's been...actual years since I moved to Discord :smile:

Miniontoby commented:
Because most bots (and mostly mine, since I haven't really seen anyone else make Minetest compatible IRC bots) just use the minetest regex for allowed usernames, like `<([a-zA-Z0-9-_]+)>` and then like [Admin] wont fit into that regex and if i would adjust the regex, then it will say the nickname is "[Admin] Miniontoby" or "Admin: Miniontoby" or we have to change it soo much that it would take too much time for me to test every thing that it would take like 2hrs maybe before I am done...

and if we put it after the nickname, then it will see [Admin] as the first argument, ruining the ! commands, since they only check for ! as start of the first argument... 

But yeah... if you can come up with a way, we could add it and test it and then if it works use it
